localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -570,8 +570,12 @@ def _resolve_refs_recursively(stack_name, resources, value: dict | list | str |
             return result
 
         if stripped_fn_lower == "findinmap":
-            attr = resolve_refs_recursively(stack_name, resources, value[keys_list[0]][1])
-            result = resolve_ref(stack_name, resources, value[keys_list[0]][0], attribute=attr)
+            attribute = value[keys_list[0]][1]
+            attr = resolve_refs_recursively(stack_name, resources, attribute)
+            selected_map = value[keys_list[0]][0]
+            if isinstance(selected_map, dict):
```

Comment:
Is it worth being more specific here, i.e
```suggestion
            if isinstance(selected_map, dict) and "Ref" in selected_map:
```